tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers_test.ts

```diff
@@ -93,4 +98,398 @@ describe('Debugger reducers', () => {
       });
     });
   });
+
+  it('Overrides activeRunId on debuggerRunsLoaded', () => {
+    const state = createDebuggerState();
+    const nextState = reducers(
+      state,
+      actions.debuggerRunsLoaded({
+        runs: {
+          __default_debugger_run__: {
+            start_time: 222,
+          },
+        },
+      })
+    );
+    expect(nextState.activeRunId).toEqual('__default_debugger_run__');
+  });
+
+  it('Updates load state on numExecutionsRequested', () => {
+    const state = createDebuggerState({
+      runs: {
+        __default_debugger_run__: {
+          start_time: 111,
+        },
+      },
+      runsLoaded: {
+        state: DataLoadState.LOADED,
+        lastLoadedTimeInMs: 222,
+      },
+      activeRunId: '__default_debugger_run__',
+    });
+    const nextState = reducers(state, actions.numExecutionsRequested());
+    expect(nextState.executions.numExecutionsLoaded.state).toEqual(
+      DataLoadState.LOADING
+    );
+    expect(
+      nextState.executions.numExecutionsLoaded.lastLoadedTimeInMs
+    ).toBeNull();
+  });
+
+  it('Updates states correctly on numExecutionsLoaded', () => {
+    const state = createDebuggerState({
+      runs: {
+        __default_debugger_run__: {
+          start_time: 111,
+        },
+      },
+      runsLoaded: {
+        state: DataLoadState.LOADED,
+        lastLoadedTimeInMs: 222,
+      },
+      activeRunId: '__default_debugger_run__',
+    });
+    let nextState = reducers(state, actions.numExecutionsRequested());
```

Comment:
I find this to be awkward. Do you need to fire another action beforehand in order to set up the environment in order to test the action? Is it possible to create the right debugger state instead?

Reply:
Yes. Done.